h/static/styles/partials-v2/_form-input.scss

```diff
@@ -50,6 +50,38 @@
     color: $grey-5;
   }
 
+  .form-input__character-counter {
+    @include font-normal;
+
+    // Show fallback "counter" if loading js times out.
+    .env-js-timeout & {
+      display: initial;
+    }
+
+    // Immediately hide counter if js capable - prevents momentary flash of
+    // unenhanced fallback counter before js kicks in.
+    .env-js-capable & {
+      display: none;
+    }
+
+    // The js controller adds this class when it's ready - show the now
+    // enhanced counter.
+    &.is-ready {
+      display: initial;
+    }
+
+    position: absolute;
+    bottom: 10px;
+    right: $h-padding;
+
+    color: $grey-5;
+  }
+
+  .is-too-long {
```

Comment:
State classes should never be used on their own, always in combination with the class they modify, eg. `.form-input__character-counter.is-too-long`. Otherwise the `is-too-long` state here will apply to any component that happens to use a state with the same name (!)